testing/test_multicall.py

```diff
@@ -184,3 +184,79 @@ def m2():
     with pytest.raises(exc):
         MC([m2, m1], {})
     assert out == ["m1 init", "m1 finish"]
+
+
+def test_hookwrapper_teardown_exception():
```

Comment:
I would adjust this name.
Consider `_handle_wrapper_unwind_in_teardown_on_exc` or something more explicit related to *where* in the chain the exception is raised.